examples/bond/assets/bond-dapp-stateful.py

```diff
@@ -0,0 +1,209 @@
+from pyteal import *
+
+def approval_program():
+    """
+    A stateful smart contract which will store the bond parameters:
+    issue price, nominal price, maturity date and coupon value, total amount.
+    Additionally, the App will store a reference to the currently
+    tradeable bond ASA and the current epoch number.
+    """
+
+    store_manager = Bytes("store_manager")
+    issue_price = Bytes("issue_price")
+    nominal_price = Bytes("nominal_price")
+    maturity_date = Bytes("maturity_date")
+    coupon_value = Bytes("coupon_value")
+    issuer_address = Bytes("issuer_address")
+    epoch = Bytes("epoch")
+    total = Bytes("total")
+    current_bond = Bytes("current_bond")
+    max_amount = Bytes("max_amount")
+    creator = Bytes("creator")
+
+    # verify rekey_to and close_rem_to are set as zero_address
+    basic_checks = And(
+        # Always verify that the RekeyTo property of any transaction is set to the ZeroAddress
+        # unless the contract is specifically involved ina rekeying operation.
+        Txn.rekey_to() == Global.zero_address(),
+        Txn.close_remainder_to() == Global.zero_address(),
+        Txn.asset_close_to() == Global.zero_address()
+    )
+
+    # initialize variables
+    on_initialize = Seq([
+        App.globalPut(store_manager, Txn.application_args[0]),
+        App.globalPut(issue_price, Btoi(Txn.application_args[1])),
+        App.globalPut(nominal_price, Btoi(Txn.application_args[2])),
+        App.globalPut(maturity_date, Btoi(Txn.application_args[3])),
+        App.globalPut(coupon_value, Btoi(Txn.application_args[4])),
+        App.globalPut(epoch, Btoi(Txn.application_args[5])),
+        App.globalPut(current_bond, Btoi(Txn.application_args[6])),
+        App.globalPut(max_amount, Btoi(Txn.application_args[7])),
+        App.globalPut(creator, Txn.application_args[8]),
+        Return(Int(1))
+    ])
+
+    # only store manager can update issuer
+    update_issuer = Seq([
+        Assert(
+             And(
+                Txn.sender() == App.globalGet(store_manager),
+                basic_checks
+            )
+        ),
+        App.globalPut(issuer_address, Txn.application_args[1]),
+        Return(Int(1))
+    ])
+
+    # only bond app manager can update issue price
+    on_update_issue_price = Seq([
```

Comment:
Could you please add in doc strings that what are the expected arguments?

Reply:
yes